### PR TITLE
feat: new decorator prop for `ComposedModal`

### LINKED ISSUE #17990: Decorator: ComposedModal

### PR DESCRIPTION
**Pull request by Gururajj77**
Closes #17990 

Updates Composed Modal to include new decorator prop

#### Changelog

**New**

- new decorator prop, classes, and styles

**Changed**

- deprecated slug prop
- update rendering slug to render the decorator component but still set the size if it's an AILabel
- updated tests

#### Testing / Reviewing

Check dropdown with AILabel storybook and form with AILabel storybook (Composed Modal)

TODO:

- [x] remove ComposedModal's "With Decorator" story before merging!

### CODE REVIEW

File: packages/react/src/components/ComposedModal/ComposedModal.tsx [hunk 1]
```diff
@@ -422,12 +433,15 @@ const ComposedModal = React.forwardRef<HTMLDivElement, ComposedModalProps>(
     }, [open, selectorPrimaryFocus, isOpen]);
 
     // Slug is always size `sm`
```
Comment (ariellalgilmore) [inline packages/react/src/components/ComposedModal/ComposedModal.tsx]:
```suggestion
    // AILabel is always size `sm`
```

File: packages/react/src/components/ComposedModal/ComposedModal.tsx [hunk 2]
```diff
@@ -422,12 +433,15 @@ const ComposedModal = React.forwardRef<HTMLDivElement, ComposedModalProps>(
...
-    }
+    let normalizedDecorator = useMemo(() => {
+      const element = slug ?? decorator;
+      if (element && element['type']?.displayName === 'AILabel') {
+        return React.cloneElement(element as React.ReactElement<any>, {
+          size: 'sm',
+        });
+      }
+      return isValidElement(element) ? element : null;
+    }, [slug, decorator]);
```
Comment (ariellalgilmore) [inline packages/react/src/components/ComposedModal/ComposedModal.tsx]: i'm wondering if we need to use `useMemo` here
```suggestion
    let normalizedDecorator = React.isValidElement(slug ?? decorator)
        ? (slug ?? decorator)
        : null;
    if (
    normalizedDecorator &&
    normalizedDecorator['type']?.displayName === 'AILabel'
    ) {
    normalizedDecorator = React.cloneElement(
      normalizedDecorator as React.ReactElement<any>,
      {
        size: 'sm',
      }
    );
  }
```

**Review by ariellalgilmore — COMMENTED** [review]
looking good just two small things

File: packages/react/src/components/ComposedModal/ComposedModal.tsx (hunk 2 above)
Comment (Gururajj77) [reply packages/react/src/components/ComposedModal/ComposedModal.tsx]: I don't think we would need, as decorator isn't something that would change too often.

**Review by ariellalgilmore — APPROVED** [review]
looks great!! 🎉

**Review by alisonjoseph — APPROVED** [review]
LGTM!

**Comment by alisonjoseph** [comment]
This is ready to merge after removing the test story and fixing the format ci check

**Review by riddhybansal — APPROVED** [review]

**Review by annawen1 — APPROVED** [review]

**Pull request merged**